packages/providers/src/lib/push/one-signal/one-signal.providerV2.spec.ts

```diff
@@ -0,0 +1,71 @@
+import { expect, test, vi, describe, beforeEach, Mocked } from 'vitest';
+import axios from 'axios';
+import { IPushOptions } from '@novu/stateless';
+import { OneSignalPushProvider } from './one-signal.provider';
+
+vi.mock('axios');
+
+const mockNotificationOptions: IPushOptions = {
+  title: 'Test',
+  content: 'Test push',
+  target: ['userId'],
+  payload: {
+    sound: 'test_sound',
+  },
+  subscriber: {},
+  step: {
+    digest: false,
+    events: [{}],
+    total_count: 1,
+  },
+};
+
+describe('test onesignal notification user api', () => {
+  const mockedAxios = axios as Mocked<typeof axios>;
+
+  beforeEach(() => {
+    mockedAxios.create.mockReturnThis();
+  });
+
+  test('should trigger OneSignal library correctly with select version', async () => {
+    const provider = new OneSignalPushProvider({
+      appId: 'test-app-id',
+      apiKey: 'test-key',
+      apiVersion: 'externalId',
+    });
+
+    const response = {
+      data: {
+        id: 'result',
+      },
+    };
+
+    mockedAxios.request.mockResolvedValue(response);
+
+    const spy = vi.spyOn(provider, 'sendMessage');
+
+    const res = await provider.sendMessage(mockNotificationOptions, {
+      iosBadgeCount: 2,
+    });
+    expect(mockedAxios.request).toHaveBeenCalled();
+    const data = JSON.parse(
+      (mockedAxios.request.mock.calls[0][0].data as string) || '{}',
+    );
+
+    expect(data).toEqual({
+      include_aliases: {
+        external_id: ['userId'],
+      },
```

Comment:
```suggestion
      include_aliases: {
        external_id: ['userId'],
      },
      target_channel: 'push',
```

I just noticed in the docs that we would need to include the target_channel for the new API endpoint:

<img width="1149" alt="image" src="https://github.com/user-attachments/assets/c30b026b-e572-4828-a2cd-d898ae8814bd">